build.gradle

```diff
@@ -41,9 +41,19 @@ apply plugin: 'com.palantir.external-publish'
 apply from: 'gradle/versions.gradle'
 
 group = 'com.palantir.atlasdb'
-version = System.env.CIRCLE_TAG ?: gitVersion()
+version = System.env.CIRCLE_TAG ?: sanitizeVersion()
 description = 'Transactional distributed database layer'
 
+def sanitizeVersion() {
+    def isClean = "git status --porcelain".execute().text.isEmpty()
+    def pluginReportedVersion = gitVersion()
+
+    if (isClean && pluginReportedVersion.endsWith(".dirty")) {
+        return pluginReportedVersion.dropRight(".dirty".length())
+    }
+    return pluginReportedVersion
+}
```

Comment:
Is this still prone to semantic merge conflicts which cause develop to be dirty? From my read of it, it is prone to that.

Reply:
It should be fine. The `isClean` check should fail if we actually end up generating something